ntge-core/src/message/decryptor.rs

```diff
@@ -0,0 +1,60 @@
+use hkdf::Hkdf;
+use secrecy::ExposeSecret;
+use sha2::Sha256;
+use x25519_dalek::StaticSecret;
+
+use crate::{
+    aead,
+    message::{self, encryptor::Encryptor},
+    x25519::FileKey,
+};
+
+#[derive(Debug)]
+pub struct Decryptor<'a> {
+    pub message: &'a message::Message,
+}
+
+impl<'a> Decryptor<'a> {
+    pub fn new(message: &'a message::Message) -> Self {
+        Decryptor { message }
+    }
+}
+
+impl<'a> Decryptor<'a> {
+    pub fn verify_message_mac(&self, file_key: &FileKey) -> bool {
```

Comment:
Should we verify the mac using `verify_message_mac` in `decrypt_payload`

Reply:
The function name describes what it does. The `decrypt_payload` just decrypt the cipher payload. Call the `verify_message_mac` if the user wants to verify the mac of the message.

The internal interface may not handle it. So the app could handle it flexible.